v2/apiserver/internal/core/projects.go

```diff
@@ -240,6 +243,36 @@ func (p *projectsService) Delete(ctx context.Context, id string) error {
 		return errors.Wrapf(err, "error retrieving project %q from store", id)
 	}
 
+	// Locate all events associated with this project
+	eventList, err := p.eventsStore.List(
+		ctx,
+		EventsSelector{ProjectID: id, WorkerPhases: WorkerPhasesAll()},
+		meta.ListOptions{},
+	)
+	if err != nil {
+		return errors.Wrapf(
+			err,
+			"error listing events associated with project %q from store",
+			id,
+		)
+	}
+
+	// Delete all events associated with this project
+	for _, event := range eventList.Items {
+		if err = p.eventsStore.Delete(ctx, event.ID); err != nil {
+			return errors.Wrapf(err, "error deleting event %q from store", id)
+		}
```

Comment:
Instead of iterating over all the events and deleting them individually, can we make a single call to `p.eventsStore.DeleteMany(...)`?

Reply:
You are right, thank you!  My thinking cap was MIA this Friday :)  Updated.